samples/bluetooth/broadcast_audio_sink/src/main.c

```diff
@@ -202,15 +202,15 @@ static void reset(void)
 	}
 }
 
-void main(void)
+int main(void)
 {
 	struct bt_audio_stream *streams_p[ARRAY_SIZE(streams)];
 	int err;
 
 	err = init();
 	if (err) {
 		printk("Init failed (err %d)\n", err);
-		return;
+		return 0;
```

Comment:
Wouldn't it make more sense to return the error? 
```suggestion
		return err;
```

This applies many places

Reply:
It might, but Zephyr explicitly ignores the return value, so it really doesn't matter in practice.

Reply:
If we ignore the return value, then why even bother adding the return type `int`? 

If we run these samples as Linux applications or part of CI, we definitely don't want to ignore the return value of the `main` function, as we want to catch errors. 

IMO blindly returning `0` in all cases is incorrect.

Reply:
Totally aligned with your remarks; it seems like an arbitrary change that has zero benefit and the potential for confusion. I've tried to clarify some of this in the doc updates; perhaps some additional wording would be helpful?

The reason for the switch is because the C and C++ standards *require* that `main` be defined as either  `int main(void)` or `int main(int argc, char **argv)` for "hosted" environments. Right now, the GCC and Clang compiler configurations include special compiler flags (`-Wno-main` and `-Wno-main-return-type`) to suppress the compiler warnings. In the future, when the -ffreestanding flag gets removed, the Clang warning turns into a Clang fatal error.

Feel free to complain to the ISO C and C++ committees about their standards, or to the Clang and GCC compiler writers for enforcing them.

Alternatively, maybe we should change the Zephyr OS so that the return value from main has some meaning?

Reply:
@keith-packard I guess it will be too much effort for you to change the `return 0` to `return err` where applicable? 

> Alternatively, maybe we should change the Zephyr OS so that the return value from main has some meaning?

As mentioned above, it does have some meaning already in certain cases

Reply:
> If we ignore the return value, then why even bother adding the return type int?

Standard conformance -- some compilers refuse to compile `void main(void)` because that is not one of the allowed types for the `main` function defined by the C standard.

> Alternatively, maybe we should change the Zephyr OS so that the return value from main has some meaning?

The return value could be used by the Zephyr kernel to do something useful (e.g. reboot if the main function returns a non-zero value with the assumption that the application initialisation failed); but, that is beyond the scope of this PR and can be addressed separately if we deem something like that to be necessary in the future.

> If we run these samples as Linux applications or part of CI, we definitely don't want to ignore the return value of the `main` function, as we want to catch errors.

But can we currently run these samples as Linux applications? If not, I do not see how that argument is valid at this time -- the return values can be updated accordingly when the sample is updated to be able to run as a Linux application, or when the Zephyr kernel is updated to actually make use of the return value -- what Zephyr kernel will do with the return value is still an open question and beyond the scope of this PR as described above.

Reply:
> But can we currently run these samples as Linux applications?

Yes :) In Nordic we do in fact also run the samples as part of CI, but given that we haven't "needed" the return values yet (we just parse the stdout data), it's a fair argument to not add those in this PR. 

If we don't add the (proper) return values in this PR, I'll just add them later for the Bluetooth samples myself :)